### PR TITLE
[13.0][MIG] mail_activity_partner

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Standard migration of module mail_activity_partner to v13.

cc ~ @ForgeFlow

### CODE REVIEW

**Review by CasVissers-360ERP — APPROVED** [review]
Functional review, only thing missing for me is a smartbutton on res.partner to activities. But this would be an enhancement of the current module.

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by HviorForgeFlow** [comment]
@OCA/social-maintainers can you move this forward?

File: mail_activity_partner/models/mail_activity.py [hunk 1]
```diff
@@ -23,14 +23,16 @@ class MailActivity(models.Model):
...
+        for activity in self:
+            res_model = activity.res_model
+            res_id = activity.res_id
+            activity.partner_id = False
+            if res_model:
+                if res_model == "res.partner":
+                    activity.partner_id = res_id
                 else:
-                    obj.partner_id = None
+                    res_model_id = self.env[res_model].search([("id", "=", res_id)])
```
Comment (simahawk) [inline mail_activity_partner/models/mail_activity.py]: why search and  not browse? In any case, if you want to search, add `limit=1`

Comment (HviorForgeFlow) [reply mail_activity_partner/models/mail_activity.py]: Seems reasonable. Comment attended.

File: mail_activity_partner/tests/test_mail_activity_partner.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Copyright 2018 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo.tests.common import TransactionCase
+
+
+class TestMailActivityPartner(TransactionCase):
+    def setUp(self):
```
Comment (simahawk) [inline mail_activity_partner/tests/test_mail_activity_partner.py]: @AdriaGForgeFlow sorry I didn't check tests before: seems you didn't see my talk on tests best practices... :stuck_out_tongue: 

Can you:
1. use SavepointCase
2. disable tracking
3. disable reset password email
?

eg:

```py
    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        # disable tracking test suite wise
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
        cls.user_model = cls.env['res.users'].with_context(no_reset_password=True)
```

Comment (AdriaGForgeFlow) [reply mail_activity_partner/tests/test_mail_activity_partner.py]: Hi @simahawk, actually I did watch the talk and found it very interesting :clap: 

The only issue is that I had forgotten about this PR at all :sweat_smile: 
I'm going to check now, thanks for the review!

Comment (AdriaGForgeFlow) [reply mail_activity_partner/tests/test_mail_activity_partner.py]: Done, can you check now? @simahawk

**Review by simahawk — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
Functional tests :+1:

**Comment by AaronHForgeFlow** [comment]
Just rebased the branch

**Pull request merged**